src/Environment.js

```diff
@@ -73,28 +69,35 @@ export default class Environment extends Relay.Environment {
       },
     });
 
-    const networkLayer = this._storeData.getNetworkLayer();
-    const subscriptionObservable = networkLayer.sendSubscription(
-      new SubscriptionRequest(query),
+    const onPayload = (payload) => {
+      updateStoreData(this, subscription.getConfigs(), query, payload);
+    };
+
+    let subscriptionObserver;
+    if (observer) {
+      const definedObserver = observer; // Placate flow.
+      subscriptionObserver = {
+        onNext: (payload) => {
+          onPayload(payload);
+          definedObserver.onNext(payload);
+        },
+        onError: (error) => {
+          definedObserver.onError(error);
+        },
+        onCompleted: () => {
+          definedObserver.onCompleted();
```

Comment:
I don't want to force the user to define the whole observer definition. e.g If I only want to listen to `onError` I should not need to define `onCompleted` or `onNext`